### PR TITLE
feat($browser): Added support for Server Side Rendering (Angular Univ…

### PR DESCRIPTION
**Pull request by bojidaryovchev**
…ersal)

Added support for Server Side Rendering (Angular Universal). The major difference is that now, upon
build, the browser code goes to dist/browser and the server code goes to dist/server, respectively.
However, its up to the backend team to implement SSR on the server.

The only breaking change is the directory structure: browser code used to go in /dist, now there are
two folders - /dist/browser and /dist/server

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -15,7 +18,8 @@ import { AppComponent } from './app.component';
 @NgModule({
   declarations: [AppComponent],
   imports: [
-    BrowserModule,
+    BrowserModule.withServerTransition({ appId: 'rxjs-dosc' }),
```
Comment (luillyfe) [inline src/app/app.module.ts]: I think it appId should be 'rxjs-docs'.

**Comment by JWO719** [comment]
Is there any difference to #173?

File: src/app/app.module.ts (hunk 1 above)
Comment (bojidaryovchev) [reply src/app/app.module.ts]: right, a typo for sure

**Comment by bojidaryovchev** [comment]
Well, as I see, #173 has additional changes which are not really necessary given the fact we only deal with the front end.. So I think this one is cleaner and doesn't deal with things we are not supposed to

**Comment by JWO719** [comment]
To be honest I don't know how to proceed, because @feloy  already did a lot of work with his pr. @ashwin-sureshkumar , @ladyleet , @sumitarora , @btroncone  any suggestions how to proceed?

**Comment by ladyleet** [comment]
Reviewing with team - will get back soon to this.

**Comment by bojidaryovchev** [comment]
nevermind, I closed it, apologies

**Pull request closed without merging**